### ISSUE TITLE
[java] New Rule AvoidConstantOnlyNames: Discourage naming such as nmbr or nmrl where vowels are omitted

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2919]
**Proposed Rule Name:**
AvoidConsonantOnlyNames

**Proposed Category:**  Code Style 

**Description:**
Discourage SMS naming such as nmbr or nmrl where vowels are omitted.

> You definitely want to discourage SMS sort of naming such as nmbr or nmrl where vowels are omitted. Or should we simply have a separate rule for this?

I personally would reject such naming.  However, if it is commonly used and descriptive enough, I can be persuaded to add these to the PMD word list.

_Originally posted by @numeralnathan in https://github.com/pmd/pmd/issues/2868#issuecomment-727213210_

**Code Sample:** 

```java
var jmh;
int tfd;

```

**Possible Properties:**
1) Minimum applicable length. Should this be a property? The minimum applicable length would be 2. I'm inclined to have it fixed at 2 to avoid names like `tm`, `nm` etc.
2) Class, interfaces,enums,method, fields and local variables.
Default to fields and local variables.
3) Customisable list of exceptions [(regexed)](https://github.com/pmd/pmd/issues/2927#issuecomment-727594039)
E.g., msg, tmp, txt.

*Should this rule be customizable via properties?*

Yes, see above.

**Comment by Fernal73** [comment]
We can also have another rule that disallows vowels only in names.

**Comment by adangel** [comment]
@Fernal73 Can you tell me, what you mean by "SMS"? I'm not sure, what you mean by "SMS naming". Is SMS just an example of poor naming?

**Comment by Fernal73** [comment]
> @Fernal73 Can you tell me, what you mean by "SMS"? I'm not sure, what you mean by "SMS naming". Is SMS just an example of poor naming?

Proper English would term it as English used while texting which omits vowels. SMS - Short Messaging Service.

**Comment by adangel** [comment]
ok, SMS as in "writing a SMS=text", as an example, where one often uses abbreviations. Thanks for clarification. I'll remove it from the title though, as it is confusing (we are not writing SMS, we are programming....).

**Comment by Fernal73** [comment]
Yes, agreed. But texting habits can creep in. Old habits die hard, best not
to form them!

**Comment by Fernal73** [comment]
Merged rule.
